src/Cli/Microsoft.DotNet.Cli.Utils/DotDefaultPathCorrector.cs

```diff
@@ -15,25 +15,22 @@ public static class DotDefaultPathCorrector
         public static void Correct()
         {
             var pathEditor = new WindowsRegistryEnvironmentPathEditor();
-            var dotDefaultPath =
-                pathEditor.Get(
-                    SdkEnvironmentVariableTarget.DotDefault);
+            var dotDefaultPath = pathEditor.Get(SdkEnvironmentVariableTarget.DotDefault);
             if (NeedCorrection(dotDefaultPath, out var correctedPath))
             {
-                pathEditor.Set(correctedPath,
-                    SdkEnvironmentVariableTarget.DotDefault);
+                pathEditor.Set(correctedPath, SdkEnvironmentVariableTarget.DotDefault);
             }
         }
 
-        internal static bool NeedCorrection(string existingPath, out string correctedPath)
+        internal static bool NeedCorrection(string? existingPath, out string correctedPath)
         {
             correctedPath = string.Empty;
             if (string.IsNullOrWhiteSpace(existingPath))
             {
                 return false;
             }
 
-            IEnumerable<string> paths = existingPath.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
+            IEnumerable<string> paths = existingPath!.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
*! Justification*: `string.IsNullOrEmpty` check

Reply:
I've decided to switch the `string.IsNullOrEmpty` check to explicit null and Length check to avoid the usage of `!`.